docs/dev/data_collector.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Data Collector
+---
+
+# Data Collector RFC
+
+The Data Collector design and API is covered in:
+
+https://github.com/chef/chef-rfc/blob/master/rfc077-mode-agnostic-data-collection.md
```

Comment:
I'd assume this repo is going to go away in the near future and dump the contents here if they're still valid

Reply:
Also that doc still talks about audit mode and really needs to be brought here so that it's current to the technology we've built / killed in the last 3-4 years